gh_pages/_source/session8/Motion-Planning-with-Tesseract.rst

```diff
@@ -0,0 +1,435 @@
+
+Prepare New Workspace:
+----------------------
+We will create a new workspace, since this exercise does not overlap with the previous exercises.
+
+#. Disable automatic sourcing of your previous workspace in your ``.bashrc`` if you had any.
+
+    .. Note:: This means you'll need to manually source the setup file from your new colcon workspace in each new terminal window.
+
+#. Source ROS 2 into your environment
+
+    .. code-block:: bash
+
+		source /opt/ros/foxy/setup.bash
+
+	.. Note:: If you are using a VM provided by us, please skip to step 6 as a directory called ``tesseract_ws`` should already exist with the dependencies have already installed for you.
+
+#. Copy the template workspace layout and files:
+
+    .. code-block:: bash
+
+    	mkdir -p ~/tesseract_ws/src
+    	cp -r ~/industrial_training/exercises/8.0/template ~/tesseract_ws/src
+    	cd ~/tesseract_ws/
+
+#. Install ``taskflow`` from the ROS-I PPA
+	.. code-block:: bash
+
+		sudo add-apt-repository ppa:ros-industrial/ppa
+		sudo apt-get update
+		sudo apt-get install taskflow
+
+#. Install the ROS 2 dependencies (this may take a while)
+	.. code-block:: bash
+
+		cd ~/tesseract_ws
+		vcs import < src/dependencies_tesseract.repos
+		vcs import < src/dependencies.repos
+		vcs import src < src/snp_automate_2022/dependencies.repos
+		rosdep install --from-paths src --ignore-src -r -y
+
+#. Initialize and Build this new workspace (this may take a little while)
+
+    .. code-block:: bash
+
+		cd ~/tesseract_ws
+		colcon build --cmake-args -DTESSERACT_BUILD_FCL=OFF
+
+#. Source the workspace
+
+    .. code-block:: bash
+
+    	source ~/tesseract_ws/install/setup.bash
+
+#. Import the new workspace into your QTCreator IDE (if using QTCreator):
+
+   * In QTCreator: `File -> New File or Project -> Other Project -> ROS Workspace -> ~/tesseract_ws`
+   * Don't forget to check the ROS distro and the build system!
+
+Intro (Review Existing Code)
+----------------------------
+Most of the infrastructure for a ROS node has already been completed for you; the focus of this exercise are Tesseract planner profiles and taskflow generators. You will notice that many files and packages are already provided for you. You could run the application as is, but you may get errors. At this time we will explore the source code that has been provided. The following are highlights of what is included.
+
+#. ``snp_automate_2022/config/worcell_plugins.yaml``:
+	* This file contains all of the kinematic plugins and contact manager plugins for our application. A kinematic plugin configuration file like this is required to use Tesseract. Take a look at ``workcell.srdf`` to see how it gets incorporated into the project.
+
+#. ``snp_motion_planning/src/planner_profiles.hpp``:
+	* This file contains the planner profiles used to create our motion plan. Currently, only the Simple Planner profile is populated. This is one of the main files we will be editing in our exercise.
```

Comment:
this will change slightly with the adjustment to planner_profiles.hpp that enables it to build